### PR TITLE
(#1761519) travis: move to CentOS 8 docker images

### PR DESCRIPTION
**Pull request by mrc0mmand**
As the CentOS 8 Docker images is finally out, we can use it and drop the
plethora of workarounds we had to implement to compile RHEL8 systemd on
CentOS 7.

---

I'll create a BZ and link it to the commit(s) once everything works as expected.

### CODE REVIEW

**Comment by mrc0mmand** [comment]
The syslog-related commit fixes following issue:

```
==3822==ERROR: AddressSanitizer: dynamic-stack-buffer-overflow on address 0x7ffda5f137c3 at pc 0x7fd4a8c24796 bp 0x7ffda5f13740 sp 0x7ffda5f13730
READ of size 1 at 0x7ffda5f137c3 thread T0
    #0 0x7fd4a8c24795 in syslog_parse_priority ../../../../src/basic/syslog-util.c:28
    #1 0x55dafd149dcc in server_process_syslog_message ../../../../src/journal/journald-syslog.c:336
    #2 0x55dafd12600d in fuzz_journald_processing_function ../../../../src/fuzz/fuzz-journald.h:28
    #3 0x55dafd126124 in LLVMFuzzerTestOneInput ../../../../src/fuzz/fuzz-journald-syslog.c:8
    #4 0x55dafd12646a in main ../../../../src/fuzz/fuzz-main.c:34
    #5 0x7fd4a8357812 in __libc_start_main (/lib64/libc.so.6+0x23812)
    #6 0x55dafd125a4d in _start (/build/build/test/fuzz/sanitize-address-fuzzers/fuzz-journald-syslog+0x17a4d)
Address 0x7ffda5f137c3 is located in stack of thread T0
SUMMARY: AddressSanitizer: dynamic-stack-buffer-overflow ../../../../src/basic/syslog-util.c:28 in syslog_parse_priority
Shadow bytes around the buggy address:
  0x100034bda6a0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100034bda6b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100034bda6c0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100034bda6d0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100034bda6e0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x100034bda6f0: 00 00 00 00 ca ca ca ca[03]cb cb cb cb cb cb cb
  0x100034bda700: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100034bda710: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 f1 f1
  0x100034bda720: f1 f1 04 f2 f2 f2 f2 f2 f2 f2 00 f2 f2 f2 f2 f2
  0x100034bda730: f2 f2 00 f2 f2 f2 f2 f2 f2 f2 00 f2 f2 f2 f2 f2
  0x100034bda740: f2 f2 00 f2 f2 f2 f2 f2 f2 f2 00 00 f2 f2 f2 f2
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==3822==ABORTING
```

**Comment by mrc0mmand** [comment]
The `sd-bus` commit fixes following issue:

```
==12871==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x60300000005c at pc 0x7f295008b59d bp 0x7ffc95e4e620 sp 0x7ffc95e4ddc8
READ of size 6 at 0x60300000005c thread T0
    #0 0x7f295008b59c  (/lib64/libasan.so.5+0xae59c)
    #1 0x7f294f9a6f09 in free_and_strndup ../../../../src/basic/string-util.c:1042
    #2 0x7f294fa2afe0 in bus_message_parse_fields ../../../../src/libsystemd/sd-bus/bus-message.c:5117
    #3 0x7f294fa0852e in bus_message_from_malloc ../../../../src/libsystemd/sd-bus/bus-message.c:562
    #4 0x55d497865a94 in LLVMFuzzerTestOneInput ../../../../src/fuzz/fuzz-bus-message.c:32
    #5 0x55d497866106 in main ../../../../src/fuzz/fuzz-main.c:34
    #6 0x7f294f309812 in __libc_start_main (/lib64/libc.so.6+0x23812)
    #7 0x55d49786556d in _start (/build/build/test/fuzz/sanitize-address-fuzzers/fuzz-bus-message+0x156d)
0x60300000005c is located 0 bytes to the right of 28-byte region [0x603000000040,0x60300000005c)
allocated by thread T0 here:
    #0 0x7f29500cc768 in __interceptor_malloc (/lib64/libasan.so.5+0xef768)
    #1 0x7f294f8c41ed in memdup ../../../../src/basic/alloc-util.c:15
    #2 0x55d4978659b8 in LLVMFuzzerTestOneInput ../../../../src/fuzz/fuzz-bus-message.c:30
    #3 0x55d497866106 in main ../../../../src/fuzz/fuzz-main.c:34
    #4 0x7f294f309812 in __libc_start_main (/lib64/libc.so.6+0x23812)
SUMMARY: AddressSanitizer: heap-buffer-overflow (/lib64/libasan.so.5+0xae59c) 
Shadow bytes around the buggy address:
  0x0c067fff7fb0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c067fff7fc0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c067fff7fd0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c067fff7fe0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c067fff7ff0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x0c067fff8000: fa fa 00 00 00 06 fa fa 00 00 00[04]fa fa fa fa
  0x0c067fff8010: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c067fff8020: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c067fff8030: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c067fff8040: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0c067fff8050: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==12871==ABORTING
```

**Comment by mrc0mmand** [comment]
Originally, I wanted to enable the clang/llvm jobs as well, but that would require significant amount of backports (like https://github.com/systemd/systemd/pull/11031/commits), so let's leave it for a separate PR.

File: ci/travis-centos-rhel8.sh [hunk 1]
```diff
@@ -95,38 +92,27 @@ for phase in "${PHASES[@]}"; do
...
+            # Upgrade the container to get the most recent environment
+            $DOCKER_EXEC dnf -y upgrade
+            # Install systemd's build dependencies
+            $DOCKER_EXEC dnf -q -y --enablerepo "PowerTools" builddep systemd
             ;;
         RUN)
             info "Run phase"
             # Build systemd
-            docker exec -it -e CFLAGS='-g -O0 -ftrapv' $CONT_NAME meson build "${CONFIGURE_OPTS[@]}"
+            docker exec -it -e CFLAGS='-g -O1 -ftrapv' $CONT_NAME meson build "${CONFIGURE_OPTS[@]}"
```
Comment (jsynacek) [inline ci/travis-centos-rhel8.sh]: I don't think that the -O0 should be changed. IMHO the tests should be run either with -O0, or the same options as we use in the spec. Maybe even both, but that's probably out of scope for this PR.

Is there a reason for the -O1 change?

Comment (mrc0mmand) [reply ci/travis-centos-rhel8.sh]: That was my mistake, the `-O1` has to be in the ASan/UBSan job due to https://github.com/systemd/systemd/issues/12997, will fix it asap.

Comment (mrc0mmand) [reply ci/travis-centos-rhel8.sh]: Ah, my past me already tried to do that, but failed, see https://github.com/systemd/systemd-centos-ci/pull/144#issuecomment-510182334. I guess I'll just revert the -Ox change, as we have gcc 8.x in RHEL8, which doesn't trigger this issue, and be done with it.

**Review by jsynacek — APPROVED** [review]
LGTM.

**Pull request merged**